lib/fs.js

```diff
@@ -1620,22 +1621,58 @@ function lstat(path, options = { bigint: false }, callback) {
 /**
  * Asynchronously gets the stats of a file.
  * @param {string | Buffer | URL} path
- * @param {{ bigint?: boolean; }} [options]
+ * @param {{ bigint?: boolean, signal?: AbortSignal }} [options]
  * @param {(
  *   err?: Error,
  *   stats?: Stats
  *   ) => any} callback
  * @returns {void}
  */
-function stat(path, options = { bigint: false }, callback) {
+function stat(path, options, callback) {
   if (typeof options === 'function') {
     callback = options;
     options = kEmptyObject;
+  } else if (options === null || typeof options !== 'object') {
+    options = kEmptyObject;
+  } else {
+    options = getOptions(options, { __proto__: null, bigint: false, signal: undefined });
```

Comment:
Non blocking but do we need a null prototype object here? It's gonna be slower

In other places with `getOptions`, there usually isn't a null proto object:

```js

  options = getOptions(options, {
    encoding: 'utf8',
    mode: 0o666,
    flag: 'w',
    flush: false,
  });
``` 

In fact I believe this will be the first instance of it

```suggestion
    options = getOptions(options, { bigint: false, signal: undefined });
```

Reply:
yes I agree with what you say, thank you for the improvement, I wonder if @jazelly  will comment

Reply:
Good catch, I was suggesting that for safeness against prototype pollution. After second look, actually, do we even need the second arg? In this case, the `typeof options` can only be 'object', which makes `defaultOption` take no effect when `getOptions`. So I think you can simply do 
```
options = getOptions(options);
```

Reply:
thanks for comments